src/algorithm/mixed_radix.rs

```diff
@@ -93,16 +97,16 @@ impl<T: FFTnum> FFTAlgorithm<T> for MixedRadix<T> {
 /// This struct is the same as MixedRadixSingle, except it's specialized for the case where both inner FFTs are butterflies
 pub struct MixedRadixDoubleButterfly<T> {
     width: usize,
-    width_size_fft: ButterflyEnum<T>,
+    width_size_fft: Rc<FFTButterfly<T>>,
```

Comment:
For the same reason that MixedRadix takes a Rc instead of a box: It's so that the planner can cache instances of it. Check out plan.rs line 42, where it checks if we already have an instance of that butterfly before making a new one.

I'm trying to think of reasons why it should be Box. One that I can come up with is lower memory overhead if butterfly instances aren't frequently re-used. what do you think?